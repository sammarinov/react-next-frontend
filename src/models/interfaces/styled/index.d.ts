import 'styled-components';

declare module 'styled-components' {
  export interface DefaultTheme {
    colors: any;
    rem: Function;
  }
}
