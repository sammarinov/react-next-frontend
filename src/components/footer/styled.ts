import styled from 'styled-components';

export default styled.footer`
  display: grid;
  background: ${({ theme }) => theme.colors.BLACK_27};
  color: ${({ theme }) => theme.colors.WHITE};
`;
