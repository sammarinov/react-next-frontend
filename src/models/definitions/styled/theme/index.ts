import { DefaultTheme } from 'styled-components';
import colors from '@Config/colors.json';

export const basePixel = 16;
export const remFn = (pixel: number): string => `${pixel / basePixel}rem`;

const theme: DefaultTheme = {
  colors,
  rem: (pixels: number | number[]) => {
    if (typeof pixels === 'number') {
      return remFn(pixels);
    }
    return pixels.map(pixel => remFn(pixel)).join(' ');
  },
};

export { theme };
