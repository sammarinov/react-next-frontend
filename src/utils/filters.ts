import { IFeatured, IArticle } from "@Redux/app/type";

export const filterFeatured = (featured: Array<IFeatured>, limit: number = -1) => {
    featured.sort((a, b) => {
        if (a.topFeatured && !b.topFeatured) return 1;
        if (!a.topFeatured && b.topFeatured) return -1;
        return parseInt(a.price) - parseInt(b.price);
    })

    if (limit > 0) return featured.slice(0, limit);
    return featured;
}

export const filterArticles = (articles: Array<IArticle>, limit: number = -1) => {
    articles.sort((a, b) => {
        return Date.parse(a.created) - Date.parse(b.created);
    })

    if (limit > 0) return articles.slice(0, limit);
    return articles;
}