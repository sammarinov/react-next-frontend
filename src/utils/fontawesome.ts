import { library } from '@fortawesome/fontawesome-svg-core';

import {
    faArrowRight
} from '@fortawesome/free-solid-svg-icons';

import {
    faTwitter,
    faFacebook,
    faInstagram,
    faYoutube
} from '@fortawesome/free-brands-svg-icons';

library.add(
    faArrowRight,
    faTwitter,
    faFacebook,
    faInstagram,
    faYoutube
)