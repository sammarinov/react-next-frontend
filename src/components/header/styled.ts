import styled from 'styled-components';

export const StyledLogo = styled.img`
  height: 70px;
`;

export default styled.header`
  display: grid;
  align-items: center;
  background: ${({ theme }) => theme.colors.BLACK_27};
  color: ${({ theme }) => theme.colors.WHITE};
  height: 80px;
`;