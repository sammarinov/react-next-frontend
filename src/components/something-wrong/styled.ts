import styled from 'styled-components';

export default styled.h1`
  font-weight: 800;
  color: #000;
  text-align: center;
  font-size: 2.5em;
  padding-top: 20px;

  @media screen and (max-width: 400px) {
    padding-left: 20px;
    padding-right: 20px;
    font-size: 2em;
  }
`;

export const StyledButtonWrapper = styled.div`
  height: 40px;
  width: 50%;
  margin: 0 auto 20px;
`;
