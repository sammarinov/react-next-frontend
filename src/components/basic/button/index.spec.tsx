import * as React from 'react';

import { render, fireEvent } from '@Test/utils';
import { theme } from '@Definitions/styled/theme';
import { Button } from '.';

describe('Basic Components', () => {
  describe('Button', () => {
    it('should have the passed class', () => {
      const { container } = render(<Button className="active" />);

      expect(container.firstChild).toHaveClass('active');
    });

    it('should render with children prop', () => {
      const { getByText } = render(
        <Button bgColor="WHITE" textColor="BLACK_27" padless curved>
          hey
        </Button>,
      );

      expect(getByText('hey')).toBeTruthy();
    });

    it('should render with children prop inverted', () => {
      const { getByText } = render(
        <Button bgColor="WHITE" textColor="BLACK_27" padless curved inverted>
          hey
        </Button>,
      );

      expect(getByText('hey')).toBeTruthy();
    });

    it('should render with multiple children', () => {
      const { getByText } = render(
        <Button bgColor="RED" curved columns="0,50" inverted>
          <span>hey</span>
          <span>hellow</span>
        </Button>,
      );

      expect(getByText('hey')).toBeTruthy();
      expect(getByText('hellow')).toBeTruthy();
    });

    it('should increment number on click', () => {
      let number = 1;
      const { getByText } = render(
        <Button
          uppercase
          leftAligned
          inverted
          onClick={() => {
            number += 1;
          }}
        >
          Button
        </Button>,
      );

      fireEvent.click(getByText('Button'), new MouseEvent('click'));

      expect(number).toBe(2);
    });

    it('should match snapshot', () => {
      const { container } = render(<Button>Test</Button>);

      expect(container).toMatchSnapshot();
    });

    it('should be disabled', () => {
      const { container } = render(
        <Button disabled sharp bgColor="RED" columns="0,1">
          Test
        </Button>,
      );

      expect(container.firstChild).toHaveStyleRule('cursor', 'not-allowed');
    });

    it('should be enabled', () => {
      const { container } = render(<Button disabled={false}>Test</Button>);

      expect(container.firstChild).toHaveStyleRule('cursor', 'pointer');
    });

    it('should have PRIMARY color', () => {
      const { container } = render(<Button>Test</Button>);

      expect(container.firstChild).toHaveStyleRule('color', theme.colors.WHITE);
    });
  });
});
