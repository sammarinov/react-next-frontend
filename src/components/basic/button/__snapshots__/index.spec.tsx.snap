// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic Components Button should match snapshot 1`] = `
.c0 {
  width: 100%;
  height: 100%;
  min-width: 1.25rem;
  min-height: 1.25rem;
  border: 1px solid transparent;
  outline: none;
  display: grid;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0.3125rem;
  grid-template-columns: 1fr;
  cursor: pointer;
  border-radius: 0.3125rem;
  background-color: #05a8aa;
  border-color: transparent;
  color: #ffffff;
}

.c0:focus {
  outline: none;
}

<div>
  <button
    class="c0"
  >
    Test
  </button>
</div>
`;
